### PR TITLE
Set up Jenkins pipeline for building and publishing Docker image

### PR DESCRIPTION
**Pull request by bisgardo**
## Purpose

Build and publish the Docker image from Jenkins for deployment in Kubernetes.

## Changes

Added Jenkinsfile containing the pipeline and made the following changes to the Dockerfile:

- Make base image a build variable and default to Node v. 16 ("latest") as that's the currently maintained LTS version. Also, using the "slim" variant shaves the image size down by almost 300 MB.
- Include `yarn.lock` in the context for stable dependencies.
- Log to stdout as that's where we pick up the logs in K8s.

The submodule URL was changed to HTTPS to make recursive checkout over HTTPS work.

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the hard-to-understand areas.

### CODE REVIEW

File: Dockerfile
Comment (Thahara) [inline Dockerfile:1]: Was this agreed on with the dev team?

File: Dockerfile
Comment (Thahara) [inline Dockerfile:7]: Why change the default to 172.17.0.1, or is the 17 a mistype?

File: Dockerfile
Comment (bisgardo) [reply Dockerfile:1]: No it's a suggestion because the existing version is too old. I tested the build with the new image. Will wait for their approval before merging.

File: Dockerfile
Comment (bisgardo) [reply Dockerfile:7]: It's because that's the default IP of the host. I think this is the only reasonable default to use in a Docker context.

**Review by shjortConcordium — APPROVED** [review]
LGTM

File: Dockerfile
Comment (shjortConcordium) [reply Dockerfile:1]: I think we just default to using 14.16. Maybe Jakob had a good reason, though.

Comment (bisgardo) [reply Dockerfile:1]: Feel free to revert the change if that's the case.

**Pull request merged**
